### ISSUE TITLE
Consider returning a value instead of throwing UnsolvedSymbolException

### ISSUE THREAD

**Issue by ftomassetti** [javaparser/javasymbolsolver#135]
This is because people seem not to be happy with seeing this exception. Sometimes a symbol cannot be solved because some dependencies are not configured. In that case the user would probably prefer some value representing an error, but not an exception. We could have to "bubble up" those errors, specifying that a certain method cannot be solved because the class cannot be solved and so on.

**Comment by ghost** [comment]
Has this been thought about recently? When analyzing large projects with >80000 LoC I get 100k+ exceptions which are simply way too many. Performance tanks significantly because of that and even
methods of JavaParserFacade like isSolved() and isReferenceType() throw exceptions which is kind of silly, imo. 

Both, Expressions and Declarations, need to return something instead of throwing Exceptions everywhere

**Comment by matozoid** [comment]
Maybe some more `Optional` pollution can be applied like in JavaParser? ;-)

**Comment by ftomassetti** [comment]
Agreed that `isSolved` should not return errors. PRs are always welcome to correct silly situations :)
SymbolReference is similar to an Optional: it can be solved or unsolved. If it is solved it contains the declaration.

**Comment by Symbolk** [comment]
Same error here, is this done or not after one year?

**Comment by matozoid** [comment]
I guess not then. Do send a PR. I'll move this issue to the active project.
